src/frontend/components/UI/Sidebar/index.tsx

```diff
@@ -34,8 +30,65 @@ export default React.memo(function Sidebar() {
     }
   }, [])
 
+  useEffect(() => {
+    if (!sidebarEl.current) return
+
+    if (sidebarSize < 120) {
+      sidebarEl.current.classList.add('collapsed')
+    } else {
+      sidebarEl.current.classList.remove('collapsed')
+    }
+
+    sidebarEl.current.style.setProperty('--sidebar-width', `${sidebarSize}px`)
+  }, [sidebarEl])
+
+  const handleDrag = (e: React.DragEvent<HTMLDivElement>) => {
+    if (e.clientX !== 0) {
+      mouseDragX = e.clientX
+    }
+  }
+
+  const handleDragStart = (e: React.DragEvent<HTMLDivElement>) => {
+    mouseDragX = e.clientX
+    dragging = true
+
+    const dragFrame = () => {
+      if (!sidebarEl.current) return
+
+      let newWidth = mouseDragX
+      if (newWidth < 60) {
+        newWidth = 60
+      } else if (newWidth > 350) {
+        newWidth = 350
+      }
+
+      if (sidebarSize !== newWidth) {
+        sidebarSize = newWidth
+
+        if (sidebarSize < 120) {
+          sidebarEl.current.classList.add('collapsed')
+        } else {
+          sidebarEl.current.classList.remove('collapsed')
+        }
+
+        sidebarEl.current.style.setProperty('--sidebar-width', `${newWidth}px`)
+      }
+
+      if (dragging) {
+        requestAnimationFrame(dragFrame)
+      }
+    }
+
+    requestAnimationFrame(dragFrame)
+  }
+
+  const handleDragEnd = () => {
+    dragging = false
+    localStorage.setItem('sidebar-width', sidebarSize.toString())
+  }
+
   return (
-    <aside className={classNames('Sidebar', { collapsed: sidebarCollapsed })}>
+    <aside ref={sidebarEl} className="Sidebar">
```

Comment:
`collapsed` class is added with javascript directly instead of using react for it to avoid rerenders